feat(ZNTA-1154):Create a cache stats page for Infinispan

broccolinisoup opened the pull request:
https://zanata.atlassian.net/browse/ZNTA-1154

The details of the branch:

    * Use infinispan as a cache manager
    * Create cache stats table on the administration side
    * Do 'clear cache' and 'clear all caches' actions
    * Enable to internalize cache stats' properties names.

Requires https://github.com/zanata/zanata-server/pull/1183

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/zanata/zanata-server/1194)
<!-- Reviewable:end -->

broccolinisoup commented:
Could you review it please @seanf @carlosmunoz ?

carlosmunoz commented:
Reviewed 17 of 17 files at r1, 2 of 6 files at r2, 1 of 2 files at r3.
Review status: 19 of 22 files reviewed at latest revision, 7 unresolved discussions.

---

*[zanata-war/src/main/java/org/zanata/action/CacheAction.java, line 46 \[r3\]](https://reviewable.io:443/reviews/zanata/zanata-server/1194#-KJmOmOP3UtquQS_gKmj:-KJmOmOP3UtquQS_gKmk:-1903237707) ([raw file](https://github.com/zanata/zanata-server/blob/e5849b3d8b757194ce8469068ca077aea9e186af/zanata-war/src/main/java/org/zanata/action/CacheAction.java#L46)):*
> ```Java
> @ViewScoped
> @Slf4j
> public class CacheAction implements Serializable {
> ```

Since this is an admin only screen, it's usually a good idea to decorate the class with `@Role('admin')`. See other Admin related beans for reference.

---

*[zanata-war/src/main/java/org/zanata/action/CacheAction.java, line 62 \[r3\]](https://reviewable.io:443/reviews/zanata/zanata-server/1194#-KJmLuRdXloEQvKSPkHP:-KJmLuRdXloEQvKSPkHQ:186559506) ([raw file](https://github.com/zanata/zanata-server/blob/e5849b3d8b757194ce8469068ca077aea9e186af/zanata-war/src/main/java/org/zanata/action/CacheAction.java#L62)):*
> ```Java
>     }
> 
>     public ArrayList<String> getCacheList() {
> ```

I would make this return a generic `List` rather than `ArrayList`. This way we can swap implementations without callers even knowing it.

---

*[zanata-war/src/main/java/org/zanata/action/CacheAction.java, line 70 \[r3\]](https://reviewable.io:443/reviews/zanata/zanata-server/1194#-KJmMYN74OUFf-v7O6lX:-KJmMYN8ITSriS4XuQwf:1725961674) ([raw file](https://github.com/zanata/zanata-server/blob/e5849b3d8b757194ce8469068ca077aea9e186af/zanata-war/src/main/java/org/zanata/action/CacheAction.java#L70)):*
> ```Java
>     public void clearCache(String cacheName) {
>         cacheManager.getCache(cacheName).clear();
>         getStats(cacheName).reset();
> ```

Not sure this line is necessary. When the cache gets cleared, don't the states for the cache get cleared too?

---

*[zanata-war/src/main/java/org/zanata/action/CacheAction.java, line 74 \[r3\]](https://reviewable.io:443/reviews/zanata/zanata-server/1194#-KJmN3m-TnLDh7c0mh1G:-KJmN3m-TnLDh7c0mh1H:1527472958) ([raw file](https://github.com/zanata/zanata-server/blob/e5849b3d8b757194ce8469068ca077aea9e186af/zanata-war/src/main/java/org/zanata/action/CacheAction.java#L74)):*
> ```Java
> 
>     public void clearAllCaches(){
>         cacheManager.getCacheNames().forEach(this::clearCache);
> ```

:+1: nice

---

*[zanata-war/src/main/java/org/zanata/action/CacheAction.java, line 82 \[r3\]](https://reviewable.io:443/reviews/zanata/zanata-server/1194#-KJmNKYbikK17uaeXokf:-KJmNKYbikK17uaeXokg:311848011) ([raw file](https://github.com/zanata/zanata-server/blob/e5849b3d8b757194ce8469068ca077aea9e186af/zanata-war/src/main/java/org/zanata/action/CacheAction.java#L82)):*
> ```Java
>     * composing the cache statistics table at the admin site. */
> 
>     public Map<String, String> getPropertyNamesAndValues(String cacheName) {
> ```

:+1: for the method's javadoc. I would also have the method name say something about the return value. For instance: `getCacheStatsProperties`, or even use Apache's own terminology: `describeCacheStatsx`

---

*[zanata-war/src/main/resources/messages.properties, line 1404 \[r3\]](https://reviewable.io:443/reviews/zanata/zanata-server/1194#-KJmO53E2VJLV1Qf2oNC:-KJmO53E2VJLV1Qf2oND:-289889978) ([raw file](https://github.com/zanata/zanata-server/blob/e5849b3d8b757194ce8469068ca077aea9e186af/zanata-war/src/main/resources/messages.properties#L1404)):*
> ```INI
> # The arguments of these methods are properties' names of StatsImpl class.
> jsf.cacheStats.timeSinceStart.name = Time Since Start
> jsf.cacheStats.timeSinceStart.description = @return Number of seconds since cache started.
> ```

Lets get rid of all the `@return`s
They don't need to show up in the GUI

---

*[zanata-war/src/main/webapp/admin/cachestats.xhtml, line 68 \[r3\]](https://reviewable.io:443/reviews/zanata/zanata-server/1194#-KJmOXJIsid-2snnIltH:-KJmOXJJFjq6cLOtxNcm:-1470654395) ([raw file](https://github.com/zanata/zanata-server/blob/e5849b3d8b757194ce8469068ca077aea9e186af/zanata-war/src/main/webapp/admin/cachestats.xhtml#L68)):*
> ```HTML
>                 <h:commandButton value="Clear All Caches"
>                   action="#{cacheAction.clearAllCaches}"
>                   onclick="return confirm('Do you confirm the cleaning of all caches?')"/>
> ```

Missed a String for i18n

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-server/1194)*
<!-- Sent from Reviewable.io -->

seanf commented:
*[zanata-war/src/main/java/org/zanata/action/CacheAction.java, line 70 \[r3\]](https://reviewable.io:443/reviews/zanata/zanata-server/1194#-KJmMYN74OUFf-v7O6lX:-KJmYV3DL7tBDikZJumS:-871489115) ([raw file](https://github.com/zanata/zanata-server/blob/e5849b3d8b757194ce8469068ca077aea9e186af/zanata-war/src/main/java/org/zanata/action/CacheAction.java#L70)):*
<details><summary><i>Previously, carlosmunoz (Carlos Munoz) wrote…</i></summary>
> Not sure this line is necessary. When the cache gets cleared, don't the states for the cache get cleared too?
</details>
Removing all the items in the cache isn't the same as removing the history (eg hit/miss rate), which I imagine are part of the stats. 

However, I'm not sure we do want to clear the statistics every time we clear the cache. I think it should be a separate button.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-server/1194)*
<!-- Sent from Reviewable.io -->

seanf commented:
Reviewed 8 of 17 files at r1, 2 of 6 files at r2, 1 of 2 files at r3.
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[zanata-war/src/main/java/org/zanata/action/CacheAction.java, line 82 \[r3\]](https://reviewable.io:443/reviews/zanata/zanata-server/1194#-KJmNKYbikK17uaeXokf:-KJmZ6stdpuV-Xa5Pvk1:-1054356869) ([raw file](https://github.com/zanata/zanata-server/blob/e5849b3d8b757194ce8469068ca077aea9e186af/zanata-war/src/main/java/org/zanata/action/CacheAction.java#L82)):*
<details><summary><i>Previously, carlosmunoz (Carlos Munoz) wrote…</i></summary>
> :+1: for the method's javadoc. I would also have the method name say something about the return value. For instance: `getCacheStatsProperties`, or even use Apache's own terminology: `describeCacheStatsx`
</details>
Given that the Map type is just Strings and Strings, we should also say exactly what the keys and values are (are they plain property names?). 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-server/1194)*
<!-- Sent from Reviewable.io -->

broccolinisoup commented:
Review status: 19 of 22 files reviewed at latest revision, 6 unresolved discussions.

---

*[zanata-war/src/main/java/org/zanata/action/CacheAction.java, line 46 \[r3\]](https://reviewable.io:443/reviews/zanata/zanata-server/1194#-KJmOmOP3UtquQS_gKmj:-KJmiC4cltPOaPrMytDl:424192559) ([raw file](https://github.com/zanata/zanata-server/blob/e5849b3d8b757194ce8469068ca077aea9e186af/zanata-war/src/main/java/org/zanata/action/CacheAction.java#L46)):*
<details><summary><i>Previously, carlosmunoz (Carlos Munoz) wrote…</i></summary>
> Since this is an admin only screen, it's usually a good idea to decorate the class with `@Role('admin')`. See other Admin related beans for reference.
</details>
I already added JSF event to check the admin role. But as we just talked, I put also @CheckRole("admin") for the bean checking. 

---

*[zanata-war/src/main/java/org/zanata/action/CacheAction.java, line 62 \[r3\]](https://reviewable.io:443/reviews/zanata/zanata-server/1194#-KJmLuRdXloEQvKSPkHP:-KJmhcEeZTsg3G9mG1XK:2112352817) ([raw file](https://github.com/zanata/zanata-server/blob/e5849b3d8b757194ce8469068ca077aea9e186af/zanata-war/src/main/java/org/zanata/action/CacheAction.java#L62)):*
<details><summary><i>Previously, carlosmunoz (Carlos Munoz) wrote…</i></summary>
> I would make this return a generic `List` rather than `ArrayList`. This way we can swap implementations without callers even knowing it.
</details>
Done! Got your point.

---

*[zanata-war/src/main/java/org/zanata/action/CacheAction.java, line 70 \[r3\]](https://reviewable.io:443/reviews/zanata/zanata-server/1194#-KJmMYN74OUFf-v7O6lX:-KJnLZNOuPNKx4gyY5pk:-335815735) ([raw file](https://github.com/zanata/zanata-server/blob/e5849b3d8b757194ce8469068ca077aea9e186af/zanata-war/src/main/java/org/zanata/action/CacheAction.java#L70)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary>
> Removing all the items in the cache isn't the same as removing the history (eg hit/miss rate), which I imagine are part of the stats. 
> 
> However, I'm not sure we do want to clear the statistics every time we clear the cache. I think it should be a separate button.
</details>
I divide the function into two functions and put another "reset" button for the JSF table. Clearing the cache and resetting the cache stats. They just do their names' actions.

---

*[zanata-war/src/main/java/org/zanata/action/CacheAction.java, line 82 \[r3\]](https://reviewable.io:443/reviews/zanata/zanata-server/1194#-KJmNKYbikK17uaeXokf:-KJnNslc_M-Oq9yeP1bo:-675589387) ([raw file](https://github.com/zanata/zanata-server/blob/e5849b3d8b757194ce8469068ca077aea9e186af/zanata-war/src/main/java/org/zanata/action/CacheAction.java#L82)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary>
> Given that the Map type is just Strings and Strings, we should also say exactly what the keys and values are (are they plain property names?). 
</details>
Changed the name of the method to "getCacheStatsProperties" and also added exactly what map's keys and values are on the comment lines. Do we need "Reset All Stats of All Caches" button or smt?

---

*[zanata-war/src/main/resources/messages.properties, line 1404 \[r3\]](https://reviewable.io:443/reviews/zanata/zanata-server/1194#-KJmO53E2VJLV1Qf2oNC:-KJmZWUc8dXJWmogkCqX:-499146022) ([raw file](https://github.com/zanata/zanata-server/blob/e5849b3d8b757194ce8469068ca077aea9e186af/zanata-war/src/main/resources/messages.properties#L1404)):*
<details><summary><i>Previously, carlosmunoz (Carlos Munoz) wrote…</i></summary>
> Lets get rid of all the `@return`s
> They don't need to show up in the GUI
</details>
Done!

---

*[zanata-war/src/main/webapp/admin/cachestats.xhtml, line 68 \[r3\]](https://reviewable.io:443/reviews/zanata/zanata-server/1194#-KJmOXJIsid-2snnIltH:-KJmgV_wALpniD2nKker:-1936393783) ([raw file](https://github.com/zanata/zanata-server/blob/e5849b3d8b757194ce8469068ca077aea9e186af/zanata-war/src/main/webapp/admin/cachestats.xhtml#L68)):*
<details><summary><i>Previously, carlosmunoz (Carlos Munoz) wrote…</i></summary>
> Missed a String for i18n
</details>
I tried all options but just one of them works (that is currently on the code). I am open to other options that you would recommend.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-server/1194)*
<!-- Sent from Reviewable.io -->

seanf commented:
Reviewed 3 of 3 files at r4.
Review status: all files reviewed at latest revision, 10 unresolved discussions, some commit checks failed.

---

*[zanata-war/src/main/java/org/zanata/action/CacheAction.java, line 82 \[r3\]](https://reviewable.io:443/reviews/zanata/zanata-server/1194#-KJmNKYbikK17uaeXokf:-KJrUcX2Ef13H6-vHQC-:-69290145) ([raw file](https://github.com/zanata/zanata-server/blob/e5849b3d8b757194ce8469068ca077aea9e186af/zanata-war/src/main/java/org/zanata/action/CacheAction.java#L82)):*
> Do we need "Reset All Stats of All Caches" button or smt?

No, I don't think it would get much use. Resetting stats will probably be rarer than clearing caches.

---

*[zanata-war/src/main/java/org/zanata/action/CacheAction.java, line 83 \[r4\]](https://reviewable.io:443/reviews/zanata/zanata-server/1194#-KJrThZ_t87gqvdZ5h2x:-KJrThZ_t87gqvdZ5h2y:1778864213) ([raw file](https://github.com/zanata/zanata-server/blob/ec5a6bd0980205838e26069128ccb67762d6b47b/zanata-war/src/main/java/org/zanata/action/CacheAction.java#L83)):*
> ```Java
>     }
> 
>     /*@Return the entire set of properties for which the specified bean provides a read method.
> ```

This should be:

    /**
     * Return...

ie double-star to start a javadoc comment, newline for style, and a space after the star before "Return" (also for style).

---

*[zanata-war/src/main/java/org/zanata/action/CacheAction.java, line 86 \[r4\]](https://reviewable.io:443/reviews/zanata/zanata-server/1194#-KJrTQZ5lB9j12W5bjem:-KJrTQZ6gZGw-DH6Ktb-:-1024396786) ([raw file](https://github.com/zanata/zanata-server/blob/ec5a6bd0980205838e26069128ccb67762d6b47b/zanata-war/src/main/java/org/zanata/action/CacheAction.java#L86)):*
> ```Java
>     * In this case, the bean is a stats object. The returning value is the set of StatsImpl
>     * (org.infinispan.stats.impl) class's properties. The returning map's keys are plain properties'
>     * names (e.g. getCurrentNumberOfEntries) and the map's values are statistic values (casted Long
> ```

Good, but `getCurrentNumberOfEntries` sounds like a method name, not a property name, which would be `currentNumberOfEntries`.

Technically, the Longs weren't cast to Strings. 

How about "the map's values are values from the Infinispan Stats object (converted to Strings if necessary).

---

*[zanata-war/src/main/java/org/zanata/action/CacheAction.java, line 87 \[r4\]](https://reviewable.io:443/reviews/zanata/zanata-server/1194#-KJrTdGvsICZc-eMbius:-KJrTdGwQWs6ybKM4At_:-1527969372) ([raw file](https://github.com/zanata/zanata-server/blob/ec5a6bd0980205838e26069128ccb67762d6b47b/zanata-war/src/main/java/org/zanata/action/CacheAction.java#L87)):*
> ```Java
>     * (org.infinispan.stats.impl) class's properties. The returning map's keys are plain properties'
>     * names (e.g. getCurrentNumberOfEntries) and the map's values are statistic values (casted Long
>     * values to String). This returning value is is used for composing the cache statistics table at
> ```

is is?

---

*[zanata-war/src/main/resources/messages.properties, line 1404 \[r4\]](https://reviewable.io:443/reviews/zanata/zanata-server/1194#-KJrV73znpiClL9cJ2cr:-KJrV74-Ebo5B1iyzQcQ:-302565059) ([raw file](https://github.com/zanata/zanata-server/blob/ec5a6bd0980205838e26069128ccb67762d6b47b/zanata-war/src/main/resources/messages.properties#L1404)):*
> ```INI
> jsf.cacheStats.confirmationCleaningAllCache=Do you confirm the cleaning of all caches?
> jsf.cacheStats.confirmationResettingCache=Do you confirm the resetting of the cache?
> jsf.ClearAllCache = Clear All Cache
> ```

This should be plural.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-server/1194)*
<!-- Sent from Reviewable.io -->

broccolinisoup commented:
Review status: all files reviewed at latest revision, 10 unresolved discussions, some commit checks failed.

---

*[zanata-war/src/main/java/org/zanata/action/CacheAction.java, line 82 \[r3\]](https://reviewable.io:443/reviews/zanata/zanata-server/1194#-KJmNKYbikK17uaeXokf:-KJriPHpXzTRNwFW6Bih:-271887862) ([raw file](https://github.com/zanata/zanata-server/blob/e5849b3d8b757194ce8469068ca077aea9e186af/zanata-war/src/main/java/org/zanata/action/CacheAction.java#L82)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary>
> > Do we need "Reset All Stats of All Caches" button or smt?
> 
> No, I don't think it would get much use. Resetting stats will probably be rarer than clearing caches.
</details>
Got it.

---

*[zanata-war/src/main/java/org/zanata/action/CacheAction.java, line 83 \[r4\]](https://reviewable.io:443/reviews/zanata/zanata-server/1194#-KJrThZ_t87gqvdZ5h2x:-KJrhAp320o7m1OKqJF_:-499146022) ([raw file](https://github.com/zanata/zanata-server/blob/ec5a6bd0980205838e26069128ccb67762d6b47b/zanata-war/src/main/java/org/zanata/action/CacheAction.java#L83)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary>
> This should be:
> 
>     /**
>      * Return...
> 
> ie double-star to start a javadoc comment, newline for style, and a space after the star before "Return" (also for style).
</details>
Done!

---

*[zanata-war/src/main/java/org/zanata/action/CacheAction.java, line 86 \[r4\]](https://reviewable.io:443/reviews/zanata/zanata-server/1194#-KJrTQZ5lB9j12W5bjem:-KJri7iOOvbQM96Uv64_:-499146022) ([raw file](https://github.com/zanata/zanata-server/blob/ec5a6bd0980205838e26069128ccb67762d6b47b/zanata-war/src/main/java/org/zanata/action/CacheAction.java#L86)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary>
> Good, but `getCurrentNumberOfEntries` sounds like a method name, not a property name, which would be `currentNumberOfEntries`.
> 
> Technically, the Longs weren't cast to Strings. 
> 
> How about "the map's values are values from the Infinispan Stats object (converted to Strings if necessary).
</details>
Done!

---

*[zanata-war/src/main/java/org/zanata/action/CacheAction.java, line 87 \[r4\]](https://reviewable.io:443/reviews/zanata/zanata-server/1194#-KJrTdGvsICZc-eMbius:-KJriE44EKcg3u6Pq6iT:-334437297) ([raw file](https://github.com/zanata/zanata-server/blob/ec5a6bd0980205838e26069128ccb67762d6b47b/zanata-war/src/main/java/org/zanata/action/CacheAction.java#L87)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary>
> is is?
</details>
Ups sorry.. Done!

---

*[zanata-war/src/main/resources/messages.properties, line 1404 \[r4\]](https://reviewable.io:443/reviews/zanata/zanata-server/1194#-KJrV73znpiClL9cJ2cr:-KJrWG5CXeL2TsWNd6zu:-499146022) ([raw file](https://github.com/zanata/zanata-server/blob/ec5a6bd0980205838e26069128ccb67762d6b47b/zanata-war/src/main/resources/messages.properties#L1404)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary>
> This should be plural.
</details>
Done!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-server/1194)*
<!-- Sent from Reviewable.io -->

seanf commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

<details><summary><i>Previously, broccolinisoup (arm) wrote…</i></summary>
> Could you review it please @seanf @carlosmunoz ?
</details>


---

Reviewed 3 of 3 files at r5.
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-server/1194#-:-KJshD9whlCgXPMduD_0:1417088721)*
<!-- Sent from Reviewable.io -->

carlosmunoz commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[zanata-war/src/main/webapp/admin/cachestats.xhtml, line 68 \[r3\]](https://reviewable.io:443/reviews/zanata/zanata-server/1194#-KJmOXJIsid-2snnIltH:-KK72F_PsdEz87BuAGKE:-2028141478) ([raw file](https://github.com/zanata/zanata-server/blob/e5849b3d8b757194ce8469068ca077aea9e186af/zanata-war/src/main/webapp/admin/cachestats.xhtml#L68)):*
<details><summary><i>Previously, broccolinisoup (arm) wrote…</i></summary>
> I tried all options but just one of them works (that is currently on the code). I am open to other options that you would recommend.
</details>
Hi Armagan, can you check if these would work? (instead of using `&quote;`

```
onclick="return confirm('#{msgs['jsf.cacheStats.confirmationCleaningAllCache']}')"
```

```
onclick='return confirm("#{msgs['jsf.cacheStats.confirmationCleaningAllCache']}")'
```

note that I changed the `&quote;` for either single or double quotes

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-server/1194)*
<!-- Sent from Reviewable.io -->

broccolinisoup commented:
Thanks

<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary>
> :lgtm:
</details>


---

Review status: 5 of 6 files reviewed at latest revision, 1 unresolved discussion.

---

*[zanata-war/src/main/webapp/admin/cachestats.xhtml, line 68 \[r3\]](https://reviewable.io:443/reviews/zanata/zanata-server/1194#-KJmOXJIsid-2snnIltH:-KK7Ba2tYmB1ToijMg04:-485019549) ([raw file](https://github.com/zanata/zanata-server/blob/e5849b3d8b757194ce8469068ca077aea9e186af/zanata-war/src/main/webapp/admin/cachestats.xhtml#L68)):*
<details><summary><i>Previously, carlosmunoz (Carlos Munoz) wrote…</i></summary>
> Hi Armagan, can you check if these would work? (instead of using `&quote;`
> 
> ```
> onclick="return confirm('#{msgs['jsf.cacheStats.confirmationCleaningAllCache']}')"
> ```
> 
> ```
> onclick='return confirm("#{msgs['jsf.cacheStats.confirmationCleaningAllCache']}")'
> ```
> 
> note that I changed the `&quote;` for either single or double quotes
</details>
I already tried the second one, it didn't work but the first one is working which I've never tried as it looks like quoting "#{msgs[" not "jsf.cacheStats.confirmationCleaningCache". Anyway, thanks a lot. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-server/1194#-:-KK7E0IxdN_oW-Oxiz3w:-2137399250)*
<!-- Sent from Reviewable.io -->

carlosmunoz commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

<details><summary><i>Previously, broccolinisoup (arm) wrote…</i></summary>
> Thanks
</details>


---

Reviewed 2 of 3 files at r5, 1 of 2 files at r6.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-server/1194#-:-KKBb81uBZoHCkm1fQcq:1417088721)*
<!-- Sent from Reviewable.io -->

djansen-redhat commented:
- [ ] Confirmation dialog shows "Do you confirm the cleaning of all caches?"  Should this say "clea_r_ing"?
- [ ] I see an unnamed cache, above "default" - what is that? (And similarly, what is "default"?)
- [ ] Would it be plausible to have the cache names as shortened (e.g. docStatusCache), with the full name/path as a tooltip?
- [ ] Some indication of what the difference between clear and reset is

seanf commented:
Reviewed 2 of 6 files at r2, 1 of 2 files at r3, 2 of 2 files at r8.
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[zanata-war/src/main/java/org/zanata/service/impl/TranslationStateCacheImpl.java, line 82 \[r8\]](https://reviewable.io:443/reviews/zanata/zanata-server/1194#-KKGTHW7o3CsibCwq5kx:-KKGTHW7o3CsibCwq5ky:-1765888158) ([raw file](https://github.com/zanata/zanata-server/blob/647559c45098aee483615e03b914100087a90d61/zanata-war/src/main/java/org/zanata/service/impl/TranslationStateCacheImpl.java#L82)):*
> ```Java
> 
>     public static final String TFT_VALIDATION_CACHE_NAME = BASE
>             + ".targetValidationCache";
> ```

Do these three cache names still need to be public constants?

---

*[zanata-war/src/main/java/org/zanata/service/impl/VersionStateCacheImpl.java, line 61 \[r8\]](https://reviewable.io:443/reviews/zanata/zanata-server/1194#-KKGT8YK1iD3xcgDtTrt:-KKGT8YK1iD3xcgDtTru:751820827) ([raw file](https://github.com/zanata/zanata-server/blob/647559c45098aee483615e03b914100087a90d61/zanata-war/src/main/java/org/zanata/service/impl/VersionStateCacheImpl.java#L61)):*
> ```Java
> 
>     public static final String VERSION_STATISTIC_CACHE_NAME = BASE
>             + ".versionStatisticCache";
> ```

Now that we are using CacheManager, I don't think the cache name constants need to be public any more.

---

*[zanata-war/src/main/resources/messages.properties, line 1406 \[r8\]](https://reviewable.io:443/reviews/zanata/zanata-server/1194#-KKGRvVLmZ0r7ADMaKFS:-KKGRvVLmZ0r7ADMaKFT:-169474707) ([raw file](https://github.com/zanata/zanata-server/blob/647559c45098aee483615e03b914100087a90d61/zanata-war/src/main/resources/messages.properties#L1406)):*
> ```INI
> jsf.cacheStats.confirmationResettingCache=Do you confirm the resetting of the cache?
> jsf.cacheStats.tooltipForClear= Removes all mappings from the cache.
> jsf.cacheStats.tooltipForReset= Removes the history. Reset statistics.
> ```

I would suggest this wording:

	jsf.cacheStats.confirmationClearingCache=Are you sure you want to clear the cache?
	jsf.cacheStats.confirmationClearingAllCache=Are you sure you want to clear all caches?
	jsf.cacheStats.confirmationResettingCache=Are you sure you want to reset the cache statistics? Cache entries will not be removed.
	jsf.cacheStats.tooltipForClear=Remove all entries from this cache.
	jsf.cacheStats.tooltipForReset=Reset statistics for this cache. Cache entries will not be removed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-server/1194)*
<!-- Sent from Reviewable.io -->

broccolinisoup commented:
Review status: 3 of 5 files reviewed at latest revision, 3 unresolved discussions.

---

*[zanata-war/src/main/java/org/zanata/service/impl/TranslationStateCacheImpl.java, line 82 \[r8\]](https://reviewable.io:443/reviews/zanata/zanata-server/1194#-KKGTHW7o3CsibCwq5kx:-KKGha49LFtccH05Hpn9:-1507414001) ([raw file](https://github.com/zanata/zanata-server/blob/647559c45098aee483615e03b914100087a90d61/zanata-war/src/main/java/org/zanata/service/impl/TranslationStateCacheImpl.java#L82)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary>
> Do these three cache names still need to be public constants?
</details>
of course not!

---

*[zanata-war/src/main/resources/messages.properties, line 1406 \[r8\]](https://reviewable.io:443/reviews/zanata/zanata-server/1194#-KKGRvVLmZ0r7ADMaKFS:-KKGZd8gN-bm1z4RGYTx:-1248372704) ([raw file](https://github.com/zanata/zanata-server/blob/647559c45098aee483615e03b914100087a90d61/zanata-war/src/main/resources/messages.properties#L1406)):*
> Reset statistics for this cache. Cache entries will not be removed.
Looks better. Done.

---

*[zanata-war/src/main/java/org/zanata/service/impl/VersionStateCacheImpl.java, line 61 \[r8\]](https://reviewable.io:443/reviews/zanata/zanata-server/1194#-KKGT8YK1iD3xcgDtTrt:-KKGhepLtxpoNtyc0e61:1087854280) ([raw file](https://github.com/zanata/zanata-server/blob/647559c45098aee483615e03b914100087a90d61/zanata-war/src/main/java/org/zanata/service/impl/VersionStateCacheImpl.java#L61)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary>
> Now that we are using CacheManager, I don't think the cache name constants need to be public any more.
</details>
I missed this point. You are right.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-server/1194)*
<!-- Sent from Reviewable.io -->

seanf commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 2 of 3 files at r9.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-server/1194#-:-KL0NcgeBrSQ_nI9G31M:1417088721)*
<!-- Sent from Reviewable.io -->

broccolinisoup commented:
@seanf @carlosmunoz could you please pass the ticket to QA?

anipeter commented:
@zanata-jenkins restest this please

anipeter commented:
Observations:-
- The clear and reset actions take place with proper indications & confirmation dialogs
- An unnamed cache is shown in the first row. Not sure what it denotes.
- The second row shows "default" - What does this indicate?
- Individual clear & Clear all buttons clears the data from individual row or entire rows respectively, but does not clear data completely, few specific columns are cleared.
- Reset button resets data per row completely (except time since reset column)
- Even if the statistics page is available only for admins, can we have this in bit more user-friendly manner, maybe with tooltips or better cache naming.
- Is it possible to have a better display of the statistics table.

seanf commented:
On 29 June 2016 at 02:10, anipeter <notifications@github.com> wrote:

> Observations:-
>
>    - The clear and reset actions take place with proper indications &
>    confirmation dialogs
>    - An unnamed cache is shown in the first row. Not sure what it denotes.
>    - The second row shows "default" - What does this indicate?
>
> As I understand it, this page is just showing what the stats APIs are
reporting. So we must have a cache named "" and one named "default". I
think they're used internally by Infinispan.


>
>    - Individual clear & Clear all buttons clears the data from individual
>    row or entire rows respectively, but does not clear data completely, few
>    specific columns are cleared.
>
> It's meant to empty out the cache, but that's not the same as clearing the
stats (that's what Reset is for). Obviously some things will be cleared,
like the number of entries should become zero


>
>    - Reset button resets data per row completely (except time since reset
>    column)
>
> Except the number of cache entries, I hope...


>
>    - Even if the statistics page is available only for admins, can we
>    have this in bit more user-friendly manner, maybe with tooltips or better
>    cache naming.
>
>
I thought Armagan was implementing tool-tips for the Clear/Reset buttons,
to make the distinction more obvious. Not sure if there were meant to be
tool-tips for the individual stats properties. Were there?

But there's nothing which can be done about the cache names here.  All this
page can do is report them.

(I suppose a possible enhancement would be to hard-code a list of
well-known cache names, so that caches could be displayed with helpful
descriptions, where available. But I think that would be out of scope for
now.)


>    - Is it possible to have a better display of the statistics table.
>
>
What do you have in mind? Note that the implementation is designed to be
very generic, so that if the Infinispan stats have different properties in
future they will be included in the table. That makes it difficult to
control the table layout in a fine-grained way.

I haven't seen the table layout in action recently. If the number of
columns (and their headings) is making the table width a mess, perhaps we
should try to do something about that.



>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/zanata/zanata-server/pull/1194#issuecomment-229098741>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AADTi2BEYjgvAFHOzy8oE3ciHnbZWWylks5qQUdggaJpZM4IvkW5>
> .
>



-- 
Sean Flanigan

Principal Software Engineer
Globalisation Tools Engineering
Red Hat

anipeter commented:
@seanf I agree and understand your points. Here is the screenshot of how it looks.
The only thing I feel is, can we have a better layout as more number of columns is making it more wider.
![cache statistics](https://cloud.githubusercontent.com/assets/12934287/16445569/de06aae0-3e00-11e6-845f-e609e5fa22da.png)

Do we need to wait to for getting a better display or can I go ahead verifying it as rest all looks good?

seanf commented:
For the unnamed cache "", perhaps it would be clearer to have (in italics):

*    (empty string)*


If the Clear/Reset buttons don't have tool-tips, I think we should add them.

Other than that, I think everything is fine for now. Although there are a
lot of columns to scroll through, they don't interfere with the main
function, which is to let you see the cache names, number of entries, and
clear/reset the caches without scrolling.


On 29 June 2016 at 18:24, anipeter <notifications@github.com> wrote:

> @seanf <https://github.com/seanf> I agree and understand your points.
> Here is the screenshot of how it looks.
> The only thing I feel is, can we have a better layout as more number of
> columns is making it more wider.
> [image: cache statistics]
> <https://cloud.githubusercontent.com/assets/12934287/16445569/de06aae0-3e00-11e6-845f-e609e5fa22da.png>
>
> Do we need to wait to for getting a better display or can I go ahead
> verifying it as rest all looks good?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/zanata/zanata-server/pull/1194#issuecomment-229289725>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AADTi3ndyX-MR9xcQIeIuPo0iue_f_SLks5qQiu1gaJpZM4IvkW5>
> .
>



-- 
Sean Flanigan

Principal Software Engineer
Globalisation Tools Engineering
Red Hat

seanf commented:
Reviewed 1 of 2 files at r10, 1 of 1 files at r11.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[zanata-war/src/main/webapp/admin/cachestats.xhtml, line 56 \[r11\]](https://reviewable.io:443/reviews/zanata/zanata-server/1194#-KLVF1ydMSeEpWOtD31G:-KLVF1ydMSeEpWOtD31H:-531970851) ([raw file](https://github.com/zanata/zanata-server/blob/5b93496ea7290dd412585bf1ca23984256e40147/zanata-war/src/main/webapp/admin/cachestats.xhtml#L56)):*
> ```HTML
>                   <c:choose>
>                     <c:when test="#{cacheName==''}">
>                       <h:outputText value="(empty string)" style="font-style: italic"/>
> ```

Good, but we should really do the styling with CSS, rather than a in-line style.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-server/1194)*
<!-- Sent from Reviewable.io -->

anipeter commented:
@seanf @broccolinisoup , the Cache Statistics page looks really good now with the inclusion of "(empty string)" in italics & tooltip for all clear/reset buttons.
Note: There is no tooltip for "Clear all Caches", but I feel that's fine as all the clear buttons has tooltip indicating what it does.
![cache statistics with tooltip](https://cloud.githubusercontent.com/assets/12934287/16481751/819c596a-3eca-11e6-9b59-4378fe9a711f.png)

aeng commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[zanata-war/src/main/webapp/admin/cachestats.xhtml, line 56 \[r11\]](https://reviewable.io:443/reviews/zanata/zanata-server/1194#-KLVF1ydMSeEpWOtD31G:-KLZ5crPP_3M3oRwy_4U:181415367) ([raw file](https://github.com/zanata/zanata-server/blob/5b93496ea7290dd412585bf1ca23984256e40147/zanata-war/src/main/webapp/admin/cachestats.xhtml#L56)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary>
> Good, but we should really do the styling with CSS, rather than a in-line style.
</details>
use `<i>` tag. Do we have jstl included? `ui:repeat` and `ui:fragment rendered="#{}"`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-server/1194)*
<!-- Sent from Reviewable.io -->

anipeter commented:
@broccolinisoup @seanf 
* Confirmed all caches docStatusCache, targetValidationCache, documentStatisticCache and versionStatisticCache
* Tool tips provided for clear/reset buttons and all headers.
![cache statistics dated 4th july](https://cloud.githubusercontent.com/assets/12934287/16552337/db814fec-41dd-11e6-94ac-acf7cd41cd47.png)

* Looks good to me :+1: , thus changing to verified.

anipeter commented:
Please note the "Zanata Web Editor (didnt run)" entry in the build page. 
Not sure if I have seen this before:
![build page](https://cloud.githubusercontent.com/assets/12934287/16552805/9cd76570-41e1-11e6-845a-8433cb9df708.png)

seanf commented:
> Please note the "Zanata Web Editor (didnt run)" entry in the build page. 

That's just Jenkins saying that sometimes when it builds this job, it there is another Maven module called "Zanata Web Editor". That sounds like the new module @davidmason is working on in another pull request. We would see something similar if a module had been removed.

anipeter commented:
thanks @seanf :+1:

seanf commented:
Reviewed 3 of 3 files at r13.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[zanata-war/src/main/java/org/zanata/action/CacheAction.java, line 80 \[r13\]](https://reviewable.io:443/reviews/zanata/zanata-server/1194#-KM2uI80nY7G--FuHkbe:-KM2uI80nY7G--FuHkbf:-152891333) ([raw file](https://github.com/zanata/zanata-server/blob/89e89b1821285f5b217a69014843ab8eb63fcb42/zanata-war/src/main/java/org/zanata/action/CacheAction.java#L80)):*
> ```Java
>         String emptyString = msgs.get("jsf.cacheStats.emptyString");
>         if (cacheName.isEmpty()) {
>             return "<em>(" + emptyString + ")</em>";
> ```

(In the interests of simplicity, we've decided to let `CacheAction` specify the `em` tag here, even though it might be purer to use a CSS class.)

---

*[zanata-war/src/main/java/org/zanata/action/CacheAction.java, line 83 \[r13\]](https://reviewable.io:443/reviews/zanata/zanata-server/1194#-KM2usRZAVfgAVwXSAgD:-KM2usRZAVfgAVwXSAgE:-957544195) ([raw file](https://github.com/zanata/zanata-server/blob/89e89b1821285f5b217a69014843ab8eb63fcb42/zanata-war/src/main/java/org/zanata/action/CacheAction.java#L83)):*
> ```Java
>         } else {
>             // Escape cache name in case it includes user input in future
>             return HtmlUtil.SANITIZER.sanitize(cacheName);
> ```

It just occurred to me that we don't want to *sanitise* cache names (as if they were suspicious user-supplied HTML). We should be escaping the cache names, to prevent them from being interpreted as HTML. Try this one: https://google.github.io/guava/releases/18.0/api/docs/com/google/common/html/HtmlEscapers.html

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-server/1194)*
<!-- Sent from Reviewable.io -->

broccolinisoup commented:
Review status: 4 of 5 files reviewed at latest revision, 4 unresolved discussions.

---

*[zanata-war/src/main/java/org/zanata/action/CacheAction.java, line 74 \[r3\]](https://reviewable.io:443/reviews/zanata/zanata-server/1194#-KJmN3m-TnLDh7c0mh1G:-KM725M4SH3WTz5UjLnW:908855107) ([raw file](https://github.com/zanata/zanata-server/blob/e5849b3d8b757194ce8469068ca077aea9e186af/zanata-war/src/main/java/org/zanata/action/CacheAction.java#L74)):*
<details><summary><i>Previously, carlosmunoz (Carlos Munoz) wrote…</i></summary>
> :+1: nice
</details>
It happened with Sean's help.

---

*[zanata-war/src/main/java/org/zanata/action/CacheAction.java, line 80 \[r13\]](https://reviewable.io:443/reviews/zanata/zanata-server/1194#-KM2uI80nY7G--FuHkbe:-KM72ta8O1zzDkYQCyTg:678204588) ([raw file](https://github.com/zanata/zanata-server/blob/89e89b1821285f5b217a69014843ab8eb63fcb42/zanata-war/src/main/java/org/zanata/action/CacheAction.java#L80)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary>
> (In the interests of simplicity, we've decided to let `CacheAction` specify the `em` tag here, even though it might be purer to use a CSS class.)
</details>
yes.

---

*[zanata-war/src/main/java/org/zanata/action/CacheAction.java, line 83 \[r13\]](https://reviewable.io:443/reviews/zanata/zanata-server/1194#-KM2usRZAVfgAVwXSAgD:-KM72-Nxfc12L2DzslT0:1360704328) ([raw file](https://github.com/zanata/zanata-server/blob/89e89b1821285f5b217a69014843ab8eb63fcb42/zanata-war/src/main/java/org/zanata/action/CacheAction.java#L83)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary>
> It just occurred to me that we don't want to *sanitise* cache names (as if they were suspicious user-supplied HTML). We should be escaping the cache names, to prevent them from being interpreted as HTML. Try this one: https://google.github.io/guava/releases/18.0/api/docs/com/google/common/html/HtmlEscapers.html
</details>
Done. How does it look like now?

---

*[zanata-war/src/main/webapp/admin/cachestats.xhtml, line 56 \[r11\]](https://reviewable.io:443/reviews/zanata/zanata-server/1194#-KLVF1ydMSeEpWOtD31G:-KM72DQrI627R89forc5:-874229428) ([raw file](https://github.com/zanata/zanata-server/blob/5b93496ea7290dd412585bf1ca23984256e40147/zanata-war/src/main/webapp/admin/cachestats.xhtml#L56)):*
<details><summary><i>Previously, aeng (Alex Eng) wrote…</i></summary>
> use `<i>` tag. Do we have jstl included? `ui:repeat` and `ui:fragment rendered="#{}"`
</details>
I got rid of the "jstl" library. No problem anymore. I used "em" tag for simplicity instead of css class.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-server/1194)*
<!-- Sent from Reviewable.io -->

seanf commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 1 files at r14.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-server/1194#-:-KM7A6fCZPVLXgD7ou8c:1417088721)*
<!-- Sent from Reviewable.io -->

seanf commented:
@anipeter would you mind taking another quick look? @broccolinisoup had to make a minor change in the HTML for the stats since your test.

anipeter commented:
@seanf sure. Shall check and revert

anipeter commented:
Cache statistics page looks good to me. :+1: 
![cache statistics_8th july](https://cloud.githubusercontent.com/assets/12934287/16685650/2e23f610-452a-11e6-9f79-c4196e2ac049.png)

Changing status to Verified.

broccolinisoup commented:
Thanks @anipeter

definite commented:
@zanata-jenkins

definite commented:
@zanata-jenkins retest this please

definite commented:
@zanata-jenkins retest this please
